src/mono/wasm/testassets/WasmBasicTestApp/wwwroot/main.js

```diff
@@ -78,7 +79,7 @@ try {
             break;
         case "LazyLoadingTest":
             if (params.get("loadRequiredAssembly") !== "false") {
-                await INTERNAL.loadLazyAssembly("System.Text.Json.wasm");
+                await INTERNAL.loadLazyAssembly(`System.Text.Json${assemblyExtension}`);
```

Comment:
I see that we are parametrizing the tested app further. DRY principle etc.
I'm not huge fan of that in testing, because it increases complexity of the boilerplate to the point that:
- it breaks more often and needs high maintenance
- it fogs and colludes the primary intent of the scenario under the test
- it's more difficult to read and reason about, even it provides true "fail" signal

This test app is on the edge as is, please let's merge this one and add no more complexity to it. 
Rather create another test app in a next folder.

my 2c

thoughts @maraf @lewing ?

Reply:
Parameterized to work for no-webcil case, which was broken. Separate app for that wouldn't make sense. We run all the tests with and without webcil.
Or did you mean some other part?

Reply:
I'm talking about the principle, next time we will want to parametrize it for MT, AOT, ... we have many features

Reply:
I'm more concerned about the tested app complexity than the infrastructure and helpers in the unit test suite. So the big part of this PR is OK.

Reply:
My goal is to have "testable samples". Something that user can interact with and also we run tests on it. We need to find a balance between single super app and tens of small apps that we need to maintain. Also, current WasmBasicTestApp is missing the "UI" the user can interact with.